### PR TITLE
[MXNET][NEURON][TEST][SAGEMAKER] - add neuron mxnet18 sm test

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/testrunner.py [hunk 1]
```diff
@@ -442,9 +442,14 @@ def main():
             sys.exit(pytest.main(pytest_cmd))
 
         else:
-            run_sagemaker_remote_tests(
-                [image for image in standard_images_list if not ("tensorflow-inference" in image and "py2" in image)]
-            )
+            if "neuron" in dlc_images:
+                run_sagemaker_remote_tests(
+                    [image for image in standard_images_list if ("mxnet" in image and "1.8.0" in image)]
```
Comment (saimidu) [inline test/testrunner.py:447]: Could you please remove this string-matching check for "1.8.0" in the image URIs being passed for testing?

This introduces issues when there are new neuron images built for future framework versions, and also causes code to be duplicated between the two if-else conditions every time there needs to be a change.

Instead, please move the framework version check to within the conftest.py for SM MXNet Inference tests for Neuron images, and change the implementation to `packaging.version.Version(framework_version) >= packaging.version.Version("1.8")` so that there is no code change that needs to happen for future versions.

File: test/sagemaker_tests/mxnet/inference/conftest.py [hunk 2]
```diff
@@ -168,6 +168,24 @@ def skip_eia_containers(request, docker_base_name):
         if 'eia' in docker_base_name:
             pytest.skip('Skipping eia container with tag {}'.format(docker_base_name))
 
+@pytest.fixture(autouse=True)
+def skip_neuron_containers(request, tag):
+    if request.node.get_closest_marker('skip_neuron_containers'):
```
Comment (saimidu) [inline test/sagemaker_tests/mxnet/inference/conftest.py:173]: @aws-vrnatham You will need to add this to other test functions that don't use neuron images, such as those dedicated to EIA/CPU/GPU.

File: test/testrunner.py [hunk 3]
```diff
@@ -449,9 +449,14 @@ def main():
             sys.exit(pytest.main(pytest_cmd))
 
         else:
-            run_sagemaker_remote_tests(
-                [image for image in standard_images_list if not ("tensorflow-inference" in image and "py2" in image)]
-            )
+            if "neuron" in dlc_images:
+                run_sagemaker_remote_tests(
+                    [image for image in standard_images_list if ("mxnet" in image)]
+                )
```
Comment (saimidu) [inline test/testrunner.py:455]: Just noticed that the sm test jobs for PyTorch and TF fail because there is no empty report file generated by `sm_utils.gnerate_empty_report()`. If the image isn't an MXNet image, then we will need to generate an empty report.

**Comment by aws-vrnatham** [comment]
Had merge issues. Changes were pushed using https://github.com/aws/deep-learning-containers/pull/1356
